### DISCUSSION TITLE
Unmount header cell component

### DISCUSSION THREAD

**Discussion by budeci** [TanStack/table#5282] [General]
Hey there, help me please!

if I toggle the checkbox it unmounts the header component, is this normal behavior?

https://github.com/TanStack/table/assets/10477258/b2605bb4-5b5f-4eda-92e8-5283be83590d

I put this list of checkboxes in a popover and if I click on the checkbox it closes, I'd prefer it to stay open just if I click outside...

**Comment by goatfryed** [comment]
This looks like an issue with your header render code and unstable keys. Since TanStack Table does not provide any markup, this shouldn't be an issue with the libary.
It sounds more like an issue with unstable react components. It would be interesting to take a look at where you render your headers. Do you use proper keys? Do your columns disappear during updates and reappear? Do you use functional components from dynamically created lambdas?
